src/odoc/resolver.ml

```diff
@@ -580,13 +579,21 @@ let create ~important_digests ~directories ~open_modules ~roots =
               { Named_roots.name; dir; omit })
             roots
         in
+        let directories =
+          match current_package with
+          | None -> directories
+          | Some pkg -> (
+              try List.assoc pkg page_roots :: directories
+              with _ -> directories)
```

Comment:
It would be better to catch only `Not_found` exceptions.

Reply:
In fact, the `Not_found` exception cannot be raised: There must be a directory associated to `current_package` it is checked before.
I think either remove the catch and add a comment, or keep the "`current_package_dir`" when computing the `current_package`?

Reply:
In pulling this particular thread a bit more unravelled than I'd expected. @panglesd could you please check carefully commit https://github.com/ocaml/odoc/pull/1241/commits/fa215e0f8f0fba7f8eacc641589720d030925280 to check I've not made any silly assumptions?